### PR TITLE
feat: add sam local get-env-vars command

### PR DESCRIPTION
**Pull request by jlhood**
*Issue #, if available:*

*Description of changes:*

Proposing a new sam local subcommand. Not sure if anyone else has already proposed it, but wrote up a quick design doc for it to get the conversation started.

*Checklist:*

- [x] Write Design Document
- [x] Write Prototype
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jasonmk** [comment]
Would this solve the issue of an ENV var that Refs another resource created by the template?

E.g., I ran into an issue today where I want to test a lambda function locally, but one of the things it needs is the arn of a secret stored in Secrets Manager.  My template has `SECRET_NAME: !Ref MySecret` as an environment variable on the function, but obviously it doesn't get populated in local mode.

If this addresses that issue, I would be a big fan of it.

**Comment by jlhood** [comment]
@jasonmk Yes, since this command reads an already deployed stack, all of those references have been resolved so the JSON that the command outputs would be resolved values.

File: samcli/commands/local/get_env_vars/cli.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    LOG.debug("'get-env-vars' command is called")
+
+    cfn_client = boto3.client('cloudformation')
+    lambda_client = boto3.client('lambda')
+
+    stack_resource_iterator = cfn_client.get_paginator('list_stack_resources').paginate(
+        StackName=stack_name
+    )
+
+    function_resource_summaries = [summary for response in stack_resource_iterator
```
Comment (TheSriram) [inline samcli/commands/local/get_env_vars/cli.py]: do we need to consider AWS::Serverless::Function?

Comment (jlhood) [reply samcli/commands/local/get_env_vars/cli.py]: No. This is a query on an actual deployed CloudFormation stack. `AWS::Serverless::Function` is a pseudo-resource of SAM, but the actual deployed stack will only contain `AWS::Lambda::Function` resources.

**Pull request closed without merging**